### PR TITLE
Add mkdocs

### PR DESCRIPTION
**Pull request by biggianteye**
Does what it says on the tin.

Does the guidance in `docs/index.md` make sense? Is there anything else that should be added?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,14 @@ Please see [CHANGELOG](CHANGELOG.md) for more information what has changed recen
 make build test
 ```
 
+You can test mkdocs by calling:
+
+```shell
+make docs-test
+```
+
+This will start a development server in docker on port `8080`.
```
Comment (h-bragg) [inline README.md]: `8000`

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+site_name: Project Name
+site_description: __package-description__
+site_author: __author-name__
+repo_url: https://github.com/graze/skeleton-project
+theme:
+  name: 'material'
+  custom_dir: theme
```
Comment (h-bragg) [inline mkdocs.yml]: in sprout, i changed this to `docs/theme`. And moved the `theme` directory underneath the `docs` directory.

File: mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        - name: php
+          lang: php
+          options:
+            startinline: True
+  - pymdownx.inlinehilite:
+  - pymdownx.emoji:
+  - pymdownx.escapeall:
+      hardbreak: True
+      nbsp: True
+  - pymdownx.tasklist:
```
Comment (h-bragg) [inline mkdocs.yml:32]:
```yaml
  - toc:
      permalink: true
```

is useful too. It adds the paragraph icon permalink icon next to headers.

File: mkdocs.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+site_name: Project Name
```
Comment (h-bragg) [inline mkdocs.yml]: Maybe we should have a 'Project Name' find and replace thing. We currently only have `skeleton-project` which is more a project slug. Just a thought.

**Review by h-bragg — CHANGES_REQUESTED** [review]
- Port fix
- i think putting the theme inside the `docs` folder make sense.

Other things we can add to the default `mkdocs.yml` file:
- `copyright: "Copyright &copy; 2018 Nature Delivered Ltd."`
- `docs_dir: 'docs'`

**Comment by biggianteye** [comment]
@h-bragg Done

**Review by h-bragg — APPROVED** [review]
Looks good, and works locally.

Not sure if this should be part of the template/this PR, but we should have some info on how to deploy to github pages:

Example: https://github.com/graze/sprout/blob/master/.travis.yml#L37
Also need to add a `GITHUB_TOKEN` to travis for deployment.

_should we detail how to setup the project in the docs itself?_

Also, perhaps adding markdown linting: https://github.com/graze/sprout/blob/master/Makefile#L57 and https://github.com/graze/sprout/blob/master/.markdownlint.json

**Comment by biggianteye** [comment]
Oh, I was going to add the markdown linting stuff too but forgot. Oops. Maybe it makes sense as a separate bit of work anyway.

I think we do need a bit more meta stuff. We have a block at the start of the README but that is getting unwieldy.

**Pull request merged**